(DOCSP-19878) December 02 2021 Release (#1507)
mongodb/docs-realm: source/endpoints.txt

.. _https-endpoints:

===============
HTTPS Endpoints
===============

.. default-domain:: mongodb

Introduction
------------

Many modern applications use multiple external services to handle
complex use cases, such as messaging, analytics, and data management. Those  
services often send data back by calling an HTTPS endpoint. You can create 
HTTPS endpoints in {+service+} for these external services to send data to 
your app.

Create a New HTTPS Endpoint
===========================
To create a new endpoint:

.. tabs-realm-admin-interfaces::

   .. tab::
      :tabid: ui

>       1. Click :guilabel:`HTTPS Endpoints` in the left navigation. 
> 
>       2. Click the :guilabel:`Add An Endpoint`.
> 
>       3. **TODO when button is enabled**

Author: ```suggestion
      1. Click :guilabel:`HTTPS Endpoints` in the left navigation. 

      #. Click the :guilabel:`Add An Endpoint` button.

      #. Enter a **route** for your endpoint. The route controls the
         URI suffix used to access your endpoint.
      
      #. Select the HTTP method used by your endpoint.
      
      #. Choose the function that your endpoint executes when it receives a request.
      
      #. Save the endpoint by clicking the :guilabel:`Save` button.
      
      #. Deploy your changes with the :guilabel:`Review Draft & Deploy` button in the draft
         banner at the top of the page.
```

.. _https-endpoints:

===============
HTTPS Endpoints
===============

.. default-domain:: mongodb

Introduction
------------

Many modern applications use multiple external services to handle
complex use cases, such as messaging, analytics, and data management. Those  
services often send data back by calling an HTTPS endpoint. You can create 
HTTPS endpoints in {+service+} for these external services to send data to 
your app.

Create a New HTTPS Endpoint
===========================
To create a new endpoint:

.. tabs-realm-admin-interfaces::

   .. tab::
      :tabid: ui

      1. Click :guilabel:`HTTPS Endpoints` in the left navigation. 

      #. Click the :guilabel:`Add An Endpoint` button.

      #. Enter a **route** for your endpoint. The route controls the
         URI suffix used to access your endpoint.
      
      #. Select the HTTP method used by your endpoint.
      
      #. Choose the function that your endpoint executes when it receives a request.
      
      #. Save the endpoint by clicking the :guilabel:`Save` button.
      
      #. Deploy your changes with the :guilabel:`Review Draft & Deploy` button in the draft
         banner at the top of the page.

> 

Author: ```suggestion

   .. figure:: /images/add_endpoint.png
      :alt: Add an HTTPS Endpoint in the {+ui+}
```

                based on the provided user credentials, if they don't match an 
                existing user. The authentication provider that corresponds to 
                the credentials must be enabled at the time of the request to 
                create the new user.
            * - ``disabled`` (boolean)
              - Enables (false) or disables (true) the endpoint.

      
      4. Once you've updated your app's configuration files, you push 
         them to your remote app. {+cli+} immediately deploys the updated 
         configurations on push.

         .. code-block:: bash
            
            realm-cli push --remote="<Your App ID>"

Migrate Webhooks to Endpoints
=============================

You should plan to migrate your existing {+service+} webhook to HTTPS 
endpoints. You do this through the UI, which provides a single-click, 
one-time migration:

.. note:: One-Time Migration

   Once your webhooks have been migrated to HTTPS endpoints, you can not 
   convert them back to webhooks. 

1. Click :guilabel:`HTTPS Endpoints` in the left navigation. 

2. Click the :guilabel:`Convert` button.

3. Click :guilabel:`Confirm`.

> .. figure:: /images/convert-endpoint.png

Author: ```suggestion
.. figure:: /images/confirmation_converting_webhooks.png
```
